library/common/jni/jni_import.h

```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+#include <jni.h>
+
+// This validates that the jni.h header that we include is *not* the jni.h provided by the JVM. This
+// helps ensure that the build is using a consistent jni.h header.
+#if defined(__ANDROID_API__) && defined(_JAVASOFT_JNI_H_)
```

Comment:
surprised you also need the latter here

Reply:
The way it works is that we still need a `#include <jni.h>` to bring it in from the NDK, so we expect this file to be included in both the android and JVM build. In the Android build we include the NDK JNI which uses `LIBNATIVEHELPER_INCLUDE_JNI_JNI_H_` as its header guard, so we check for `_JAVASOFT_JNI_H_` which is the header guard used by the JVM build to fail the build if we included the wrong jni.h for the android build